articles/why-you-should-use-celery-with-rabbitmq/index.md

```diff
@@ -0,0 +1,463 @@
+
+### Outline
+
+1. [What We Will Cover In This Article](#what-we-will-cover-in-this-article)
+2. [Understanding Celery & RabbitMQ](#understanding-celery-&-rabbitmq)
+    - [What is a Task Queue](#what-is-a-queue)
+    - [Examples of Task Queues](#examples-of-task-queues)
+    - [What is a Message Broker](#what-is-a-message-broker)
+    - [Examples of Message Brokers](#examples-of-message-brokers)
+3. [Why We Should Choose The Celery & RabbitMQ Combo](#why-we-should-choose-the-celery-&-rabbitmq-combo)
+4. [Setting up Celery with RabbitMQ](#setting-up-celery-with-rabbitmq)
+    - [Installation & Configuration](#installation-&-configuration)
+    - [Simple Use Case](#simple-use-case)
+5. [Conclusion](#conclusion)
+
+### What We Will Cover In This Article
+
+In this article, we’ll walk through what tasks queue are, see the examples of task queues (which includes celery), learn about message brokers. We will also look at why we need to use Celery and RabbitMQ while also seeing a basic tutorial that shows its installation and usage.
+
+### Understanding Celery & RabbitMQ
+
+Before going into why we should use Celery & RabbitMQ, let us understand what they are and what similar tools are available. Celery is classified as a *Task Queue* and RabbitMQ is classified as a *Message Broker*. Let's dive deeper into what these are.
+
+#### What is a Task Queue?
+
+A task queue is a data structure maintained by a job scheduler containing jobs to run.
+Task queue software also manages background work that must be executed outside of the usual HTTP request-response cycle. They are designed for asynchronous work.
+To further explain their usefulness, let’s say we have a web application that uses artificial intelligence to enhance images.
+As the number of users increase, the time to enhance an image drastically increases, which leads to a significant delay while enhancing.
+Hence, the need for a task queu/e software as it efficiently manages requests and ensures the application runs smoothly.
+
+#### Examples of Task Queue Software
+
+- Celery
+- Redis Queue
+- Taskmaster
+- Huey
+- tasq
+
+#### What is a Message Broker ?
+
+A message broker allows applications, systems and services to communicate and exchange information with each other. Now, with a task queue software in the background, the web app needs to know the status of the job in progress, plus any errors that might occur and the end result of the enhancement. A message broker dramatically eases this process since it’s built to make independent processes “talk to each other”
```

Comment:
```suggestion
A message broker allows applications, systems and services to communicate and exchange information with each other. Now, with a task queue software in the background, the web app needs to know the status of the job in progress, plus any errors that might occur and the end result of the enhancement. A message broker eases this process since it’s built to make independent processes “talk to each other”
```